OWML.ModHelper.Input/ModInputHandler.cs

```diff
@@ -0,0 +1,475 @@
+                && (combination.GetPressDuration() < _tapDuration);
+        }
+
+        private bool IsNewlyReleased_Single(IModInputCombination combination, bool keep = false)
+        {
+            return (!IsPressed_Single(combination)) && combination.IsFirst(keep)
+                && ((ModInputCombination)combination).IsRelevant(_tapKeep);
+        }
+
+        public bool IsPressed(IModInputCombination combination)
+        {
+            return IsPressed_Combo(combination) || IsPressed_Single(combination);
+        }
+
+        public bool IsNewlyPressed(IModInputCombination combination, bool keep = false)
+        {
+            return IsPressed(combination) && combination.IsFirst(keep);
+        }
+
+        public bool WasTapped(IModInputCombination combination)
+        {
+            return (!(IsPressed_Combo(combination) || IsPressed_Single(combination)))
+                && ((ModInputCombination)combination).IsRelevant(_tapKeep)
+                && (combination.GetPressDuration() < _tapDuration);
+        }
+
+        public bool WasNewlyReleased(IModInputCombination combination, bool keep = false)
+        {
+            return (!(IsPressed_Combo(combination) || IsPressed_Single(combination)))
+                && ((ModInputCombination)combination).IsRelevant(_tapKeep);
+        }
+
+        private KeyCode StringToKeyCode(string key)
+        {
+            if (key.Contains(xboxPrefix))
+            {
+                string xboxKey = key.Substring(xboxPrefix.Length);
+                var xboxCode = (XboxButton)Enum.Parse(typeof(XboxButton), xboxKey, true);
+                if (Enum.IsDefined(typeof(XboxButton), xboxCode))
+                {
+                    return InputTranslator.GetKeyCode(xboxCode, false);
+                }
+                else
+                {
+                    return KeyCode.None;
+                }
+            }
+            else
+            {
+                string changedKey = key;
+                if (key == "control" || key == "ctrl")
+                {
+                    return KeyCode.LeftControl;
+                }
+                else if (key == "shift")
+                {
+                    return KeyCode.LeftShift;
+                }
+                else if (key == "alt")
+                {
+                    return KeyCode.LeftAlt;
+                }
+                KeyCode code = (KeyCode)Enum.Parse(typeof(KeyCode), changedKey, true);
+                if (Enum.IsDefined(typeof(KeyCode), code))
+                {
+                    return code;
+                }
+                else
+                {
+                    return KeyCode.None;
+                }
+            }
+        }
+
+        private Int64 ParseCombination(string combo, bool forRemoval = false)
+        {
+            combo = combo.Trim();
+            int[] thisCombination = new int[_maxComboLength];
+            int i = 0;
```

Comment:
var (everywhere)

Reply:
went on rampage and put vars even in for and foreach *shrug*